internal/dag/builder_test.go

```diff
@@ -1113,6 +1246,105 @@ func TestDAGInsertGatewayAPI(t *testing.T) {
 				},
 			),
 		},
+		"insert basic single route, single hostname, gateway with TLS": {
+			gateway: gatewayWithOnlyTLS,
+			objs: []interface{}{
+				sec1,
+				kuardService,
+				genericHTTPRoute,
+			},
+			want: listeners(
+				&Listener{
+					Port: 443,
+					VirtualHosts: virtualhosts(
+						&SecureVirtualHost{
+							VirtualHost: VirtualHost{
+								Name:         "test.projectcontour.io",
+								ListenerName: "ingress_https",
+								routes:       routes(prefixroute("/", service(kuardService))),
+							},
+							Secret: secret(sec1),
+						},
+					),
+				},
+			),
+		},
+		"insert basic single route, single hostname, gateway with missing TLS certificate": {
+			gateway: gatewayWithOnlyTLS,
+			objs: []interface{}{
+				kuardService,
+				genericHTTPRoute,
+			},
+			want: listeners(),
+		},
+		"insert basic single route, single hostname, gateway with invalid TLS certificate": {
```

Comment:
this looks identical to the previous test case, seems like it should have an actual invalid secret being inserted?

Reply:
👍